app/controllers/searches_controller.rb

```diff
@@ -9,7 +11,7 @@ def new
   end
 
   def create
-    @search = Search.new(query: @query, filter: @filter)
+    @search = Search.new(adaptor: @adaptor)
```

Comment:
should this perhaps be a read_only attr? 
`(adaptor: adaptor)`

Reply:
do you mean something like this

```
def create
  @search = Search.new(adaptor: adaptor)
  ...
end

...

def adaptor
   case @filter
    when 'defendant'
      CourtDataAdaptor::Query::Defendant.new(term)
    else
      CourtDataAdaptor::Query::ProsecutionCase.new(term)
    end
  end
...

```

Reply:
This should do too, or else, you might want to make it an `attr_reader`, so as to avoid using an instance var.

Reply:
have added that now